### PR TITLE
[Feature: #163722982]Navbar Component

### PR DESCRIPTION
**Pull request by habibaudu**
#### What does this PR do?

- Create Navbar Component

#### Description of Task to be completed?

- [x] Create navBar folder
- [x] Create navBar Component
- [x] Write test

#### How should this be manually tested?

On terminal run `npm run start:dev`

#### What are the relevant pivotal tracker stories?
[#163722982](https://www.pivotaltracker.com/story/show/163722982)

### CODE REVIEW

**Comment by ebenezerdon** [comment]
Great job! @habibaudu
Is there any reason you prefer unzipping `materialize` into the project to using `cdn`?
Could you also check out [react-materialize](https://react-materialize.github.io/#/)?

**Comment by habibaudu** [comment]
> Great job! @habibaudu
> Is there any reason you prefer unzipping `materialize` into the project to using `cdn`?
> Could you also check out [react-materialize](https://react-materialize.github.io/#/)?
> Thanks @ebenezerdon 
>  I think I neither `unzipped materialize` nor `used cdn`. I used the npm package `npm install materialize-css@next` then I imported it in our App.jsx like this  `import 'materialize-css' ,
import 'materialize-css/dist/css/materialize.min.css'`
>I tried using cdn in our root index.html , it didn't work
>`React-materialize`  only implements a subset of Materialize's components, that's why i didn't use it.
>And just to provide some more clarity, The reason I used the npm package `npm install materialize-css@next` is beacause of our `stylelinting`, the `scss and css` violates most of our rules. so by using the `npm packages` we just exclude `node_modules` in our webpack_config file and our code runs.

**Comment by ebenezerdon** [comment]
Okay, I'm not quite sure why everything `materialize` is in our project's folder. Is there a way to avoid this?
It shows you added `23755` lines and 58 files.
I thought this was what the `npm` packages aim to solve.

**Comment by habibaudu** [comment]
> Okay, I'm not quite sure why everything `materialize` is in our project's folder. Is there a way to avoid this?
> It shows you added `23755` lines and 58 files.
> I thought this was what the `npm` packages aim to solve.

> No the `npm package` is not supposed to resolve that, it is not intstalling materialize as a whole,
its installing the different components (cards, navbar....etc). that's why its that large.

**Comment by andela-mabdussalam** [comment]
The npm package adds all files materialize related in node_modules(you can go ahead to check that they are all there). This suggests that they can be referenced as opposed to added directly into the application.

Your adding it here defeats the purpose of having them in the node_modules. Also you can exclude files that you do not want to lint, if that is an issue for you.

**Comment by habibaudu** [comment]
> The npm package adds all files materialize related in node_modules(you can go ahead to check that they are all there). This suggests that they can be referenced as opposed to added directly into the application.
> 
> Your adding it here defeats the purpose of having them in the node_modules. Also you can exclude files that you do not want to lint, if that is an issue for you.

> Ok noted.
> The problem was the js folder which was first generated by the zip. I have deleted it.

File: src/app/navBar/NavbarComponent.test.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+import NavbarComponent from './NavbarComponent'
+
+describe(' Component', () => {
+    it('should render the NavBar', () => {
+        const component = shallow(<NavbarComponent />)
+        expect(component.exists()).toBe(true)
+    })
+})
```
Comment (rajeman) [inline src/app/navBar/NavbarComponent.test.js]: Why is there no snapshot test?

**Comment by andela-mabdussalam** [comment]
Taking a first look at the design, a couple of things
- Subjects are not center aligned on the navbar.
- The horizontal bar between the sign up and sign in is not properly coloured
- The font weight should be more
- The size of the logo is inconsistent
- The font size should be increased

File: src/app/navBar/NavbarComponent.test.js (hunk 1 above)
Comment (habibaudu) [reply src/app/navBar/NavbarComponent.test.js]: Thanks  for spotting that.
Am on it.

**Comment by habibaudu** [comment]
> Taking a first look at the design, a couple of things
> 
> * Subjects are not center aligned on the navbar.
> * The horizontal bar between the sign up and sign in is not properly coloured
> * The font weight should be more
> * The size of the logo is inconsistent
> * The font size should be increased

> Ok. Will attend to this Asap

File: src/app/navBar/NavbarComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                    </div>
+                    <div className="hide-on-med-and-down show">
+                        <ul>
+                            <li>
+                                <a href="#!">SignIn</a>
+                            </li>
+                            <li style={{ color: '#5ed3f1' }}>|</li>
+                            <li>
+                                <a href="#!">SignUp</a>
+                            </li>
```
Comment (rajeman) [inline src/app/navBar/NavbarComponent.jsx]: Can you remove the inline styles?

Comment (habibaudu) [reply src/app/navBar/NavbarComponent.jsx]: Thanks for spotting that.
I have removed the inline style

**Comment by rajeman** [comment]
@habibaudu Please resolve the conflicts on your branch

**Comment by habibaudu** [comment]
> @habibaudu Please resolve the conflicts on your branch

Ok...
done

File: jest.config.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
 module.exports = {
-    setupFiles: ['./testSetup.js'],
-    transform: { '^.+\\.js|jsx$': 'babel-jest' },
+    setupFiles: ['<rootDir>/src/tests/setup.js'],
+    transform: { '^.+\\.jsx?$': 'babel-jest' },
     collectCoverageFrom: ['src/**/*.{js,jsx}'],
     snapshotSerializers: ['enzyme-to-json/serializer'],
     moduleNameMapper: {
-        '\\.(jpg|jpeg|png|gif|svg)$': '<rootDir>/__mocks__/fileMock.js',
+        '\\.(jpg|jpeg|png|gif|svg)$': '<rootDir>/__mocks__/filemock.js',
```
Comment (andela-mabdussalam) [inline jest.config.js]: Why is the file changed to lowercase here?

File: package.json [hunk 4]
```diff
@@ -9,7 +9,7 @@
     "serve": "node server.js",
     "start": "npm run build && npm run serve",
     "test": "jest --coverage --coverageReporters=text-lcov | coveralls",
-    "test:dev": "jest --coverage"
+    "test:dev": "jest"
```
Comment (andela-mabdussalam) [inline package.json]: Please add the `--coverage` flag back. We want to see a report when we run it locally?.

File: src/app/home/HomeContainer.js [hunk 5]
```diff
@@ -1,10 +1,8 @@
...
 
 const HomeContainer = () => {
-  return (
-    <HomeComponent />
-  );
+    return <HomeComponent />
 }
 
-export default HomeContainer;
+export default HomeContainer
```
Comment (andela-mabdussalam) [inline src/app/home/HomeContainer.js]: We want the semi colons

File: package.json (hunk 4 above)
Comment (habibaudu) [reply package.json]: ok

File: src/app/home/HomeContainer.js (hunk 5 above)
Comment (habibaudu) [reply src/app/home/HomeContainer.js]: ok

File: jest.config.js (hunk 3 above)
Comment (habibaudu) [reply jest.config.js]: I will rectify that.

File: .stylelintrc [hunk 6]
```diff
@@ -7,10 +7,13 @@
       "ignore": ["stylelint-commands", "after-comment"]
     } ],
     "declaration-colon-space-after": "always",
+    "indentation": [4, {
```
Comment (andela-mabdussalam) [inline .stylelintrc]: Did we agree on 2 space indentation?

File: package.json [hunk 7]
```diff
@@ -56,6 +56,7 @@
     "stylelint-config-standard": "^18.2.0",
     "stylelint-webpack-plugin": "^0.10.5",
     "terser": "^3.14.1",
+    "thunk": "0.0.1",
```
Comment (andela-mabdussalam) [inline package.json]: Are we using this package?

File: package.json [hunk 8]
```diff
@@ -38,7 +39,6 @@
     "enzyme-adapter-react-16": "^1.9.1",
     "express": "^4.16.4",
     "file-loader": "^3.0.1",
-    "google-fonts-webpack-plugin": "^0.4.4",
```
Comment (andela-mabdussalam) [inline package.json:41]: I am curious to know why we removed this.

File: src/app/home/HomeComponent.jsx [hunk 9]
```diff
@@ -1,9 +1,11 @@
 import React, { Fragment } from 'react';
 import '../../style/home.scss';
+import Navbar from '../navBar/NavbarComponent.jsx';
 
 const HomeComponent = () => {
   return (
     <Fragment>
+      <Navbar />
```
Comment (andela-mabdussalam) [inline src/app/home/HomeComponent.jsx]: Let us add this to `App.js`, the same way the footer is there.

File: .stylelintrc (hunk 6 above)
Comment (XROLE) [reply .stylelintrc]: I think we did agree to use indentation of 2

File: package.json (hunk 8 above)
Comment (XROLE) [reply package.json:41]: At some point the plugin became problematic and the error message was not comprehensive as such we decided to do away with it at the moment. Though we were thinking of referencig the CDN on index.html file

File: src/app/home/HomeComponent.jsx (hunk 9 above)
Comment (XROLE) [reply src/app/home/HomeComponent.jsx]: @habibaudu

**Comment by andela-mabdussalam** [comment]
@habibaudu I am also concerned about the dropped coverage, at the very least, the coverage should be the same.

File: package.json (hunk 7 above)
Comment (habibaudu) [reply package.json]: No we are not...

File: src/app/home/HomeComponent.jsx (hunk 9 above)
Comment (habibaudu) [reply src/app/home/HomeComponent.jsx]: ok

**Comment by habibaudu** [comment]
> @habibaudu I am also concerned about the dropped coverage, at the very least, the coverage should be the same.

ok
I will improve the coverage.

File: .stylelintrc (hunk 6 above)
Comment (habibaudu) [reply .stylelintrc]: ok

**Review by andela-mabdussalam — APPROVED** [review]

**Pull request merged**